smart-freight-center/sfc-backend
# chore(db): adds postgresql database to the project

## Description

- creates code structure of how the gateways would be used
- adds migrations and models for participants and participant_connection
- adds setup for Sequelize ORM to the app
- creates a boundary around all data fetching using data gateways
- I also tried to add a `GitHub` action that runs tests whenever code is pushed

### How to test it

- Add env variables for the test database(see .env.sample)
- Run integration tests(`yarn test:integration)


## Learning


- The DataGateway allows us to create an effective boundary between the database and other parts of our system. This allows us to treat the DB as a detail and separate it from the main logic of our system.  It also  makes it possible for us to replace the DB later on if we need to
